### ISSUE TITLE
chinese dataset with attn is worse than CTC

### ISSUE THREAD

**Issue by haohaoxiao-dl** [clovaai/deep-text-recognition-benchmark#107]
I used the same parameters except "Prediction", and I got 60+ accuracy with CTC while the accuracy of attn is no more than 10,than i tried an adam learning rate of 0.0003,than my training result of attn can up to 20+,which still too low.

what can I do to improve the result?

by the way, on account of the arbitrary size of my training data, I set the imgW to 512 and use "pad",and my batch_max_length was set to 64(once tried to set a bigger and a smaller one, but seems no palpable improve on accuracy)

thanks

**Comment by boy977** [comment]
@haohaoxiao-dl Is that accuracy measured on val set or test set? I got > 87% accuracy on val but less than 10% accuracy on test :(

**Comment by haohaoxiao-dl** [comment]
> @haohaoxiao-dl Is that accuracy measured on val set or test set? I got > 87% accuracy on val but less than 10% accuracy on test :(

just the accuracy on val set.
in your case, I think enlarge the training data size would work

**Comment by boy977** [comment]
@haohaoxiao-dl I use more than 1 million textboxes. That's should be enough I think

**Comment by haohaoxiao-dl** [comment]
> @haohaoxiao-dl I use more than 1 million textboxes. That's should be enough I think

Should there be any undetected difference between your test set and val set?

**Comment by boy977** [comment]
> Should there be any undetected difference between your test set and val set?

of course these are differences between synthetic data with real data but I didn't expect the result would be too poor like that

**Comment by haohaoxiao-dl** [comment]
> > Should there be any undetected difference between your test set and val set?
> 
> of course these are differences between synthetic data with real data but I didn't expect the result would be too poor like that

maybe you can fine-tune your trained model by some real data?

BTW,I tried to set Transformation to TPS,but the accuracy was always zero during training.Any suggestion about that?

**Comment by boy977** [comment]
> BTW,I tried to set Transformation to TPS,but the accuracy was always zero during training.Any suggestion about that?
sorry I'm so noob about this. IMO, the accuracy is zero means that the model does not learn anything. Usually I will check my data or try to tune the learning rate
